cdap-app-fabric/src/test/java/co/cask/cdap/ConditionalWorkflowApp.java

```diff
@@ -116,6 +128,31 @@ public void beforeSubmit(MapReduceContext context) throws Exception {
       String outputPath = context.getRuntimeArguments().get("outputPath");
       FileInputFormat.addInputPath(job, new Path(inputPath));
       FileOutputFormat.setOutputPath(job, new Path(outputPath));
+      // Put something in the token
+      context.getWorkflowToken().put("action_type", "MapReduce");
+      context.getWorkflowToken().put("start_time", String.valueOf(System.currentTimeMillis()));
+    }
+
+    // TODO: CDAP-2894 MapReduce counters should be available implicitly from the WorkflowToken.
+    // We yet to figure out the scheme.
+    @Override
+    public void onFinish(boolean succeeded, MapReduceContext context) throws Exception {
+      if (!succeeded) {
+        return;
+      }
+
+      WorkflowToken workflowToken = context.getWorkflowToken();
+      if (workflowToken == null) {
```

Comment:
nit: this is kind of ugly. apps should ideally not have to do this check. But unsure how we can fix it (if we can). Maybe lets report a jira?